crates/pallet-domains/src/migrations.rs

```diff
@@ -58,3 +58,47 @@ mod domain_balance_check_migration_migration {
         T::DbWeight::get().reads_writes(1, 1)
     }
 }
+
+pub type VersionCheckedMigrateDomainsV4ToV5<T> = VersionedMigration<
+    4,
+    5,
+    VersionUncheckedMigrateV3ToV4<T>,
+    Pallet<T>,
+    <T as frame_system::Config>::DbWeight,
+>;
+
+pub struct VersionUncheckedMigrateV4ToV5<T>(sp_std::marker::PhantomData<T>);
+impl<T: Config> UncheckedOnRuntimeUpgrade for VersionUncheckedMigrateV4ToV5<T> {
+    fn on_runtime_upgrade() -> Weight {
+        domain_genesis_receipt::set_domain_genesis_receipt::<T>()
+    }
+}
+
+mod domain_genesis_receipt {
+    use super::Config;
+    use crate::{DomainGenesisBlockExecutionReceipt, ExecutionReceiptOf};
+    use hexlit::hex;
+    use sp_core::{Get, H256};
+    use sp_domains::DomainId;
+    use sp_runtime::Weight;
+
+    pub(super) fn set_domain_genesis_receipt<T: Config>() -> Weight {
+        let genesis_state_root = T::DomainHash::from(H256::from(hex!(
+            "0x530eae1878202aa0ab5997eadf2b7245ee78f44a35ab25ff84151fab489aa334"
+        )));
+
+        let genesis_block_hash = T::DomainHash::from(H256::from(hex!(
+            "0x5a367ed131b9d8807f0166651095a9ed51aefa9aaec3152d3eb5cee322220ce6"
+        )));
+
+        let domain_0_genesis_er = ExecutionReceiptOf::<T>::genesis(
+            genesis_state_root,
+            sp_domains::EMPTY_EXTRINSIC_ROOT.into(),
+            genesis_block_hash,
+        );
+
+        DomainGenesisBlockExecutionReceipt::<T>::insert(DomainId::new(0), domain_0_genesis_er);
```

Comment:
Do we need to check we're running on the correct network before calling this migration?
Or does it work for both Taurus and Mainnet?
(We could also check it manually, but we'd have to remember to remove the migration before deploying any runtime upgrades to Mainnet.)

I'm thinking something like the check here:
https://github.com/autonomys/subspace/pull/3495/files#diff-cce0a7bce11b64c46af5b9a7b8fc4e82d4581d53cb420e46cd40ec99e3edfc4bR95

Reply:
> We could also check it manually, but we'd have to remember to remove the migration before deploying any runtime upgrades to Mainnet.

This is what I was mostly thinking about right now. Run on taurus and remove the migration before deploying to mainnet as there is some time.

Will have to check if the taurus genesis exisits on runtime actually.